src/components/download-button/download-button.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import Button from '@mapbox/mr-ui/button';
+import IconText from '@mapbox/mr-ui/icon-text';
+import PropTypes from 'prop-types';
+
+export default class DownloadButton extends React.PureComponent {
+  render() {
+    const { fileType, href } = this.props;
+    const fileName = href
+      .toString()
+      .split('/')
+      .pop()
+      .replace(/\.\w+$/, '');
+
+    if (fileType === 'CSV') {
+      return (
+        <Button href={href}>
+          <IconText iconBefore="arrow-down">Download {fileType} file</IconText>
+        </Button>
+      );
+    } else {
+      return (
+        <Button href={href} passthroughProps={{ download: fileName }}>
+          <IconText iconBefore="arrow-down">Download {fileType} file</IconText>
+        </Button>
+      );
+    }
+  }
+}
+
+DownloadButton.propTypes = {
+  href: PropTypes.string.isRequired,
+  fileType: PropTypes.oneOf(['CSV', 'PDF', 'PNG', 'ZIP']).isRequired
```

Comment:
I love an opinionated PropType!

I am wondering if we should include... more? 

- JPG
- GeoJSON
- JSON
- GIF
- HTML
- CSS
- JS


**Or, maybe the most future friendly approach is to allow any value here (`PropTypes.string.isRequired`)?** 

By using `PropTypes.string.isRequired`, we don't have to:
- maintain a list of possible file extensions
- make updates to this component
- ignore PropTypes warnings in repositories that use a `fileType` outside this list

While we lose the ability to enforce consistency, it will make the component more flexible.

**OR! What if we didn't include a `fileType` prop and extract the file type from the `href`?**

While this will add a small bit of code to the component, we:
- reduce the need for one more propType for the contributor to have to remember to add
- can restyle (letter up/down-casing) the file extension to match our style guide as needed
- make the component future friendly


What do you think?

Reply:
Ooof!

I glossed over the "Download for iOS" button (ref )https://github.com/mapbox/dr-ui/pull/474#issuecomment-902425956). I love using this button for that use case, but it will change a few of the ideas above. 

To support that use case we could have these propTypes:

- `href` (required)
- `text` (optional) 
  - If entered, use this value as the button's text. Example `text="Download for iOS"` results in `Download for iOS`. 
  - If empty, the default button text will be `Download {extension} file`. Example `Download PNG`. We could extract the extension from the `href` prop.

Reply:
> OR! What if we didn't include a fileType prop and extract the file type from the href?

Love this so much! For _"can restyle (letter up/down-casing) the file extension to match our style guide as needed"_, can you point me to the best place to find this information in the style guide? I do see ZIP referenced in the Word list -> https://hey.mapbox.com/documentation/style-guide/word-list/ 🕺 

I added an optional text prop to support the iOS use case and future download buttons that require custom text 👍

Reply:
> For "can restyle (letter up/down-casing) the file extension to match our style guide as needed", can you point me to the best place to find this information in the style guide? I do see ZIP referenced in the Word list -> https://hey.mapbox.com/documentation/style-guide/word-list/


That's exactly where I would point you to. We can also check with Heather and Dom!